ci-operator/step-registry/cucushift/installer/rehearse/azure/cases/bootstrap/provision/cucushift-installer-rehearse-azure-cases-bootstrap-provision-commands.sh

```diff
@@ -0,0 +1,153 @@
+    if [[ -f "${SHARED_DIR}/bastion_public_address" ]]; then
+        bastion_ip=$(<"${SHARED_DIR}/bastion_public_address")
+        bastion_ssh_user=$(<"${SHARED_DIR}/bastion_ssh_user")
+        ssh_proxy_command="-o ProxyCommand='ssh ${ssh_options} -W %h:%p ${bastion_ssh_user}@${bastion_ip}'"
+    fi
+
+    echo "ssh ${ssh_options} ${ssh_proxy_command} core@${node_ip} '${cmd}'" | sh -
+}
+
+if [[ -z "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" ]]; then
+  echo "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE is an empty string, exiting"
+  exit 1
+fi
+echo "Installing from release ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+check_result=0
+
+SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+cluster_name="${NAMESPACE}-${UNIQUE_HASH}"
+install_dir="/tmp/${cluster_name}"
+mkdir -p ${install_dir}
+cat "${SHARED_DIR}/install-config.yaml" > "${install_dir}/install-config.yaml"
+
+echo "Creating cluster ..."
+cat "${install_dir}/install-config.yaml" | grep -v "password\|username\|pullSecret\|auth" | tee ${ARTIFACT_DIR}/install-config.yaml
+export AZURE_AUTH_LOCATION=${CLUSTER_PROFILE_DIR}/osServicePrincipal.json
+openshift-install create cluster --dir="${install_dir}" 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
+
+try=0
+max_try=25
+while [[ ${try} -lt ${max_try} ]]; do
+    if grep -qP 'Waiting up to 45m0s.*for bootstrapping to complete...' "${install_dir}/.openshift_install.log"; then
+        echo "INFO: installer ran into stage of waiting for bootstrap to complete, break the installation..."
+        # shellcheck disable=SC2046
+        kill $(jobs -p)
+        break
+    fi
+    echo "waiting for installer running into stage of waiting for bootstrap to complete..."
+    sleep 120
+    try=$(( try + 1 ))
+done
+
+if [ X"${try}" == X"${max_try}" ]; then
+    echo "ERROR: installer does not run into bootstrap complete stage, exit!"
+    exit 1
+fi
+
+# Check that all clients on bootstrap host to localhost for k8s API access
+echo "**********Check that all clients on bootstrap host to localhost for k8s API access*********"
+infra_id=$(jq -r '.infraID' ${install_dir}/metadata.json)
+resource_group="${infra_id}-rg"
+bootstrap_public_ip_id=$(az network lb frontend-ip list --lb-name ${infra_id} -g ${resource_group} -ojson | jq -r '.[] | select(.inboundNatRules != null) | .publicIPAddress.id')
+bootstrap_public_ip=$(az network public-ip show --ids ${bootstrap_public_ip_id} --query 'ipAddress' -otsv)
+ssh_ret=$(ssh_command "${bootstrap_public_ip}" 'sudo grep "server:" /etc/kubernetes/kubeconfig')
```

Comment:
Are there multiple lines of server in the kubeconfig, or only one line ?

Reply:
Or to avoid multi servers show up in /etc/kubernetes/kubeconfig, maybe we can run:
```
ssh_ret=$(ssh_command "${bootstrap_public_ip}" 'sudo KUBECONFIG=/etc/kubernetes/kubeconfig oc config view --minify | grep "server:"'
```

Reply:
checking

Reply:
only one line in kubeconfig:
```
[root@ip-10-0-99-168 ~]# cat /etc/kubernetes/kubeconfig  | grep server:
    server: https://localhost:6443
```

Reply:
Okay. If only one line, it is okay for now, but if introduce some new config in the future, the check might return a false positive result, so maybe https://github.com/openshift/release/pull/58585/files#r1841802127 is safer, whatever, on your call now.